core/services/keeper/upkeep_executer.go

```diff
@@ -202,155 +160,38 @@ func (ex *UpkeepExecuter) processActiveUpkeeps() {
 		// in case we miss the UpkeepPerformed log or the tx errors. It does not need to be exact.
-		err = ex.orm.SetLastRunHeightForUpkeepOnJob(ctx, ex.job.ID, upkeep.UpkeepID, headNumber)
+		err := ex.orm.SetLastRunHeightForUpkeepOnJob(ctxService, ex.job.ID, upkeep.UpkeepID, headNumber)
 		if err != nil {
 			return errors.Wrap(err, "failed to set last run height for upkeep")
 		}
 
-		_, err = ex.pr.InsertFinishedRun(postgres.UnwrapGorm(postgres.TxFromContext(ctx, ex.orm.DB)), pipeline.Run{
-			State:          pipeline.RunStatusCompleted,
-			PipelineSpecID: ex.job.PipelineSpecID,
-			Meta: pipeline.JSONSerializable{
-				Val: map[string]interface{}{"eth_tx_id": etx.ID},
-			},
-			Errors: runErrors,
-			Outputs: pipeline.JSONSerializable{Val: []interface{}{
-				fmt.Sprintf("queued tx from %v to %v txdata %v", etx.FromAddress, etx.ToAddress, hex.EncodeToString(etx.EncodedPayload)),
-			}},
-			CreatedAt:  start,
-			FinishedAt: null.TimeFrom(f),
-		}, false)
-		if err != nil {
-			return errors.Wrap(err, "failed to insert finished run")
-		}
 		return nil
-	})
-	if err != nil {
-		svcLogger.WithError(err).Error("failed to update database state")
-	}
-
-	// TODO: Remove in
-	// https://app.clubhouse.io/chainlinklabs/story/6065/hook-keeper-up-to-use-tasks-in-the-pipeline
-	elapsed := time.Since(start)
-	pipeline.PromPipelineTaskExecutionTime.WithLabelValues(fmt.Sprintf("%d", ex.job.ID), ex.job.Name.String, "", job.Keeper.String()).Set(float64(elapsed))
-	var status string
-	if runErrors.HasError() || err != nil {
-		status = "error"
-		pipeline.PromPipelineRunErrors.WithLabelValues(fmt.Sprintf("%d", ex.job.ID), ex.job.Name.String).Inc()
-	} else {
-		status = "completed"
-	}
-	pipeline.PromPipelineRunTotalTimeToCompletion.WithLabelValues(fmt.Sprintf("%d", ex.job.ID), ex.job.Name.String).Set(float64(elapsed))
-	pipeline.PromPipelineTasksTotalFinished.WithLabelValues(fmt.Sprintf("%d", ex.job.ID), ex.job.Name.String, "", job.Keeper.String(), status).Inc()
-}
-
-func (ex *UpkeepExecuter) constructCheckUpkeepCallMsg(upkeep UpkeepRegistration) (ethereum.CallMsg, error) {
-	checkPayload, err := RegistryABI.Pack(
-		checkUpkeep,
-		big.NewInt(int64(upkeep.UpkeepID)),
-		upkeep.Registry.FromAddress.Address(),
-	)
-	if err != nil {
-		return ethereum.CallMsg{}, errors.Wrap(err, "failed to pack payload of check upkeep")
-	}
-
-	to := upkeep.Registry.ContractAddress.Address()
-	gasLimit := ex.config.KeeperRegistryCheckGasOverhead() + uint64(upkeep.Registry.CheckGas) +
-		ex.config.KeeperRegistryPerformGasOverhead() + upkeep.ExecuteGas
-
-	return ethereum.CallMsg{
-		From: utils.ZeroAddress,
-		To:   &to,
-		Gas:  gasLimit,
-		Data: checkPayload,
-	}, nil
-}
-
-// logRevertReason logs the given error with a log level depends on this given error context
-// Default log level is error. Mapping between a reason and its log level:
-//  - "upkeep not needed": Debug
-func logRevertReason(logger *logger.Logger, err error) {
```

Comment:
have you compared the info in the logs after these changes? Extracting the revert reason is really important to helping us debug for clients. We may need to add this to the eth_call adapter.